File: react-native-compat/ios/OfflineFaceModule.swift
import Foundation

@objc(OfflineFaceModule)
class OfflineFaceModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc(initializeModels:rejecter:)
  func initializeModels(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    // Load ONNX assets from the iOS app bundle models directory.
    resolve(nil)
  }

  @objc(getModelProfile:rejecter:)
  func getModelProfile(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    reject("NOT_WIRED", "Native ONNX runtime wiring is pending.", nil)
  }

  @objc(registerFace:frameBase64:resolver:rejecter:)
  func registerFace(
    name: String,
    frameBase64: String,
    resolve: RCTPromiseResolveBlock,
    reject: RCTPromiseRejectBlock
  ) {
    reject("NOT_WIRED", "Native ONNX runtime wiring is pending.", nil)
  }

  @objc(verifyFace:requireActiveChallenge:resolver:rejecter:)
  func verifyFace(
    frameBase64: String,
    requireActiveChallenge: Bool,
    resolve: RCTPromiseResolveBlock,
    reject: RCTPromiseRejectBlock
  ) {
    reject("NOT_WIRED", "Native ONNX runtime wiring is pending.", nil)
  }

  @objc(getPendingEvents:rejecter:)
  func getPendingEvents(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    reject("NOT_WIRED", "Local mobile event store is pending.", nil)
  }

  @objc(syncPendingEvents:rejecter:)
  func syncPendingEvents(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    reject("NOT_WIRED", "AWS sync endpoint integration is pending.", nil)
  }

  @objc(purgeSyncedEvents:rejecter:)
  func purgeSyncedEvents(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    reject("NOT_WIRED", "Purge integration is pending.", nil)
  }
}

File: react-native-compat/android/OfflineFaceModule.kt
package com.datalake.offlineface

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class OfflineFaceModule(private val context: ReactApplicationContext) :
    ReactContextBaseJavaModule(context) {

    override fun getName(): String = "OfflineFaceModule"

    @ReactMethod
    fun initializeModels(promise: Promise) {
        // Load ONNX assets from android/app/src/main/assets/models.
        promise.resolve(null)
    }

    @ReactMethod
    fun getModelProfile(promise: Promise) {
        // Return detector, recognizer, liveness model names and total bytes.
        promise.reject("NOT_WIRED", "Native ONNX runtime wiring is pending.")
    }

    @ReactMethod
    fun registerFace(name: String, frameBase64: String, promise: Promise) {
        // Decode frame, run detector + embedding, store encrypted local embedding.
        promise.reject("NOT_WIRED", "Native ONNX runtime wiring is pending.")
    }

    @ReactMethod
    fun verifyFace(frameBase64: String, requireActiveChallenge: Boolean, promise: Promise) {
        // Decode frame, run liveness + recognition, queue offline event.
        promise.reject("NOT_WIRED", "Native ONNX runtime wiring is pending.")
    }

    @ReactMethod
    fun getPendingEvents(promise: Promise) {
        promise.reject("NOT_WIRED", "Local mobile event store is pending.")
    }

    @ReactMethod
    fun syncPendingEvents(promise: Promise) {
        promise.reject("NOT_WIRED", "AWS sync endpoint integration is pending.")
    }

    @ReactMethod
    fun purgeSyncedEvents(promise: Promise) {
        promise.reject("NOT_WIRED", "Purge integration is pending.")
    }
}
